app/src/main/java/com/globant/harrypotterapp/activity/MainActivity.kt

```diff
@@ -18,33 +23,66 @@ class MainActivity : AppCompatActivity() {
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
         setOnClickListeners()
+        mainViewModel.getHousesLiveData().observe(::getLifecycle, ::updateUI)
         mainViewModel.fetchHouses()
     }
 
     private fun setOnClickListeners() {
         binding.mainActivitySpellsButton.setOnClickListener {
-            startActivity(SpellsActivity.getIntent(this))
+            mainViewModel.goToSpells()
         }
         binding.mainActivityGryffindorButton.setOnClickListener {
-            startActivity(HouseActivity.getIntent(this).putExtra(HOUSE_NAME, GRYFFINDOR))
+            mainViewModel.goToHouse(GRYFFINDOR)
         }
         binding.mainActivityHufflepuffButton.setOnClickListener {
-            startActivity(HouseActivity.getIntent(this).putExtra(HOUSE_NAME, HUFFLEPUFF))
+            mainViewModel.goToHouse(HUFFLEPUFF)
         }
         binding.mainActivityRavenclawButton.setOnClickListener {
-            startActivity(HouseActivity.getIntent(this).putExtra(HOUSE_NAME, RAVENCLAW))
+            mainViewModel.goToHouse(RAVENCLAW)
         }
         binding.mainActivitySlytherinButton.setOnClickListener {
-            startActivity(HouseActivity.getIntent(this).putExtra(HOUSE_NAME, SLYTHERIN))
+            mainViewModel.goToHouse(SLYTHERIN)
         }
     }
 
+    private fun updateUI(data: Event<MainData>) {
+        val mainState = data.getContentIfNotHandled()
+        when (mainState?.status) {
+            MainStatus.LOADING_MAIN -> {
+                showMessage(R.string.main_activity_loading_message)
+            }
+            MainStatus.SUCCESS_MAIN -> {
+                showMessage(R.string.main_activity_success_message)
+            }
+            MainStatus.ERROR_MAIN -> {
+                showMessage(R.string.main_activity_error_message)
+            }
+            MainStatus.GO_HOUSE -> {
+                openHouseActivity(mainState.houseName)
+            }
+            MainStatus.GO_SPELLS -> {
+                openSpellsActivity()
+            }
+        }
+    }
+
+    private fun showMessage(stringId: Int) {
+        Toast.makeText(this, getString(stringId), Toast.LENGTH_SHORT).show()
+    }
+
+    private fun openHouseActivity(houseName: String) {
+        startActivity(HouseActivity.getIntent(this, houseName))
+    }
+
+    private fun openSpellsActivity() {
+        startActivity(SpellsActivity.getIntent(this))
+    }
+
     companion object {
         fun getIntent(context: Context): Intent = Intent(context, MainActivity::class.java)
         private const val GRYFFINDOR = "Gryffindor"
```

Comment:
you have a constantUtils with this value, maybe you can put all the hases names there

Reply:
You are right, done!